Tryouts/Plugins/Services/MockAuthentication/MockAuthenticationService.cs

```diff
@@ -0,0 +1,75 @@
+﻿/// ********************************************************************************************************
+///
+/// Morgan Stanley makes this available to you under the Apache License, Version 2.0 (the "License").
+/// You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+/// See the NOTICE file distributed with this work for additional information regarding copyright ownership.
+/// Unless required by applicable law or agreed to in writing, software distributed under the License
+/// is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and limitations under the License.
+/// 
+/// ********************************************************************************************************
+
+using NP.Utilities;
+using NP.Utilities.Attributes;
+using NP.Utilities.BasicServices;
+
+namespace MockAuthentication;
+
+[Implements(typeof(IAuthenticationService), IsSingleton = true)]
+public class MockAuthenticationService : VMBase, IAuthenticationService
+{
+    [Part]
```

Comment:
I think both property and constructor parameter injection can be useful in different situations, so both can be implemented.

Reply:
Again we should weigh the pros with the cons and also discuss if there are scenarios that require specific options.  After @BalassaMarton mentions an objective observation of using property injection, that is valuable input into the conversation.    I would personally phrase the same observation a little bit differently and add related notes for an improvement in clarity. 
**Note**
Higher degrees of Cohesion are generally seen as a positive attribute signifying well written code.  Part of the reason that high cohesion is good is that it improves readability, it reduces the cognitive load required for understanding and it reduces incidental complexity.  The idea that high Cohesion and low Coupling is good arose within the world of structured programming.  Here they are discussed as properties of modules.  While structured programming may not purely pre-date OOP, it was definitely distinct.  Modules here aren't like the modularity we speak of in a CUIAF.  When you read about the types of cohesion in software it's useful to think of the word module as meaning "any bounded unit of code aggregation."  So at one end of the spectrum we can talk about the coupling/cohesion of a single function, at the other we can talk about the degrees of coupling/cohesion in a subsystem (like the file system of an operating system).  Some of these type of cohesion are considered more valuable than others.  

One example of cohesion within a class that people follow without realising the value of doing so ties back to cohesion:  Many coding guidelines advise that all fields in a class should be grouped together in one area in the file for that class.  In and of itself this is a form of Logical Cohesion.  They are aggregated because they all do one thing, store state.  If they are further organised into sub-groupings, then these groupings may provide further (more valuable) forms of cohesion.

**Observation**
* If you only use constructor injection then this defines a single unit of cohesion for dependencies.  (I am finding it difficult to determine exactly the type of cohesion, it is at least Logical Cohesion and could probably be argued to be Communicational Cohesion)

Reply:
I surrender on this - we can have only constructor dependencies